### PR TITLE
IC-1625: Add new enforceable days page to Refer journey and remove old RAR days page

### PR DESCRIPTION
**Pull request by Gweaton**
## What does this pull request do?

- Switches `draftReferrals` to use new `maximumEnforceableDays`, rather than `maximumRarDays` field and removes that and `usingRarDays` fields.
- Adds new page for setting maximum enforceable days.
- Removes deprecated fields on the `draftReferral` object.
- Updates all points that expected RAR days field to use `maximumEnforceableDays`

## What is the intent behind these changes?

This will replace the current `maximumRarDays` and `usingRarDays` fields, now that enforceable days is not an optional field, as PPs always have to specify a number of enforceable days.

## Screenshot

image

### CODE REVIEW

File: server/models/draftReferral.ts [hunk 1]
```diff
@@ -23,8 +23,9 @@ export interface ReferralFields {
   whenUnavailable: string | null
   serviceUser: ServiceUser
   additionalRiskInformation: string
-  usingRarDays: boolean
-  maximumRarDays: number | null
+  usingRarDays: boolean // deprecated
+  maximumRarDays: number | null // deprecated
+  maximumEnforceableDays: number | null
```
Comment (lawrence-forooghian) [inline server/models/draftReferral.ts:28]: I reckon this will no longer be nullable, now that the user has to enter a number of days?

Comment (Gweaton) [reply server/models/draftReferral.ts:28]: Doh, I was getting confused with where it starts out as `null` but that's where we extend it with nullable fields. Will fix this, thanks.

**Review by lawrence-forooghian — APPROVED** [review]

**Comment by Gweaton** [comment]
Marking as draft, as we've decided to do everything in this one PR.

**Review by tomsmyers — APPROVED** [review]
brilliant - thank you !

**Pull request merged**